boards/opentitan/src/tests/sip_hash.rs

```diff
@@ -52,9 +41,28 @@ impl<'a> SipHashTestCallback {
 }
 
 impl<'a> hasher::Client<'a, 8> for SipHashTestCallback {
-    fn add_data_done(&'a self, result: Result<(), ErrorCode>, _data: &'static mut [u8]) {
+    fn add_data_done(&'a self, result: Result<(), ErrorCode>, data: &'static mut [u8]) {
         assert_eq!(result, Ok(()));
         self.data_add_done.set(true);
+
+        let temp = self.input_save.take().unwrap();
+
+        // Stay within valid ranges
+        assert_eq!(self.cb_count.get() < 20, true);
+        for i in 0..data.len() {
+            assert_eq!(i < 8, true);
+            temp[self.cb_count.get() as usize][i] = data[i];
+        }
+
+        self.input_save.replace(temp);
```

Comment:
You aren't replacing `data` here.

Does something like this work?

```rust
SipHashTestCallback {
...
    input: [TakeCell<'static, [u8; 8]>; 20],
...
}

fn add_data_done(&'a self, result: Result<(), ErrorCode>, data: &'static mut [u8]) {
...
    self.input[i].replace(data);
...
}
```
temp[]

Reply:
Where, `input` here acts as a separate buffer to collect the callback slices? But doesn't replace `input_buffer` (that starts off with all the input data)  once all the cbs are complete?